datasource/utils.go

```diff
@@ -4,6 +4,61 @@ package datasource
 
 import "strconv"
 
+var DivisionalAreas = [3]string{"Arts and Humanities", "Social Sciences", "Science & Engineering & Applied Science"}
+
+func getGenEdInfo(intAttributes []interface{}) []string {
+
+	// Type conversion, since we're dealing with an interface.
+	var attributes []map[string]interface {}
+	for _, item := range intAttributes {
+		attrMap, ok := item.(map[string]interface {}) 
+		if !ok {
+			panic("An item in courseAttributes is not of type map[string]interface")
+		}
+		attributes = append(attributes, attrMap)
+	}
+
+	areas := []string{}
+	for _, attribute := range attributes {
+		if attribute["crseAttributeDescription"] == "FAS General Education" {
+			areas = append(areas, attribute["crseAttrValueDescription"].(string))
+		}
+	}
+	return areas // if empty, not a GENED. 
+}
+
+func checkDivisionalArea(val string) bool {
+	for _, area := range DivisionalAreas {
+		if val == area {
+			return true
+		}
+	}
+	return false
+}
+
+func getDivisionalInfo(intAttributes []interface{}) []string {
+
+	// Type conversion, since we're dealing with an interface.
+	var attributes []map[string]interface {}
+	for _, item := range intAttributes {
+		attrMap, ok := item.(map[string]interface {})
+		if !ok {
+			panic("An item in courseAttributes is not of type map[string]interface")
+		}
+		attributes = append(attributes, attrMap)
+	}
+
+	areas := []string{}
+	for _, attribute := range attributes {
+		divAttr, ok := attribute["crseAttrValueDescription"].(string)
+		if !ok {  continue  } // sometimes the value is nil. Just move to the next one.
+		if attribute["crseAttributeDescription"] == "FAS Divisional Distribution" && checkDivisionalArea(divAttr) {
```

Comment:
Are there other divisional areas with this `crseAttribute`? If so I'd probably just keep them for now, we can filter it out later in the frontend if necessary or display it otherwise.

Reply:
Inspection of 2021 and 2022 reveals that any courses that do not fall under the 3 divisional areas but have the "FAS Divisional Distribution" instead have `None` as their field for `crseAttrValueDescription` (these are atypical classes like freshman seminars/grad classes/etc that are under FAS but don't meet a divisional requirement). It seems appropriate for these to return an empty array for divisional areas (current behavior).

Reply:
Okay great!